### ISSUE TITLE
How to train a facial landmarks detection model which could detect more than 68 points?

### ISSUE THREAD

**Issue by crissallan** [davisking/dlib#1978]
Hello, I recently used dlib (the python module) for detecting human facial landmarks. I'd like to say dilb's detection is fast and pretty accurate!
However, I'd like to detect more facial landmarks points(e.g. 500 points) to improve the performance. Since dlib's facial landmark detection is based on regression trees, so I'd like to train them by my self.(don't worry for the data, my partner will help me to label some face images). 
So I'd like to know is there any way that I can modify the dlib facial landmarks detector to let it detect 500 points in a face image? And if the answer is yes, how should I train it?

**Comment by arrufat** [comment]
I would start reading the official example on how to train custom shape predictors:
- http://dlib.net/train_shape_predictor.py.html

There's a series of posts on the pyimagesearch blog that you might find interesting:
1. https://www.pyimagesearch.com/2019/12/16/training-a-custom-dlib-shape-predictor/
1. https://www.pyimagesearch.com/2019/12/23/tuning-dlib-shape-predictor-hyperparameters-to-balance-speed-accuracy-and-model-size/

**Comment by crissallan** [comment]
> I would start reading the official example on how to train custom shape predictors:
> 
> * http://dlib.net/train_shape_predictor.py.html
> 
> There's a series of posts on the pyimagesearch blog that you might find interesting:
> 
> 1. https://www.pyimagesearch.com/2019/12/16/training-a-custom-dlib-shape-predictor/
> 2. https://www.pyimagesearch.com/2019/12/23/tuning-dlib-shape-predictor-hyperparameters-to-balance-speed-accuracy-and-model-size/

Thank you so much! Those are really helpful for me!

**Comment by joel5638** [comment]
Hi @crissallan . Those links are to create custom shape predictor which are again regression tree based. Can you help me with any tensorflow based process using faster rcnn or any model?
